compiler/rustc_middle/src/ty/list.rs

```diff
@@ -58,8 +58,8 @@ impl<T> List<T> {
         // The empty slice is static and contains a single `0` usize (for the
         // length) that is 64-byte aligned, thus featuring the necessary
         // trailing padding for elements with up to 64-byte alignment.
-        static EMPTY_SLICE: InOrder<usize, MaxAlign> = InOrder(0, MaxAlign);
-        unsafe { &*(&EMPTY_SLICE as *const _ as *const List<T>) }
+        const EMPTY_SLICE: &'static InOrder<usize, MaxAlign> = &InOrder(0, MaxAlign);
+        unsafe { &*(EMPTY_SLICE as *const _ as *const List<T>) }
```

Comment:
Oh no this is lacking a very important comment (it's on the method but not this `static`).

The use of `static` is *very deliberate* because `const` would cause "multiple different empty lists", and this is used with interning + pointer comparisons.

I'm guessing that makes this PR non-viable?

Reply:
I guess so. I thought that `const EMPTY_SLICE: &'static` will have to be the same reference -- after all it's a constant, but then later was told by @oli-obk that this is not the case (I still don't understand why constants are so cursed but anyway).